src/components/ArticlesList.jsx

```diff
@@ -2,30 +2,89 @@ import { fetchArticles } from "../api.js";
 import Articles from "./Articles.jsx";
 import { useState, useEffect } from "react";
 import Loading from "./Loading.jsx";
-import {useSearchParams} from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
 
 const ArticlesList = () => {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [sortAttributes, setSortAttributes] = useState([]);
+  const [order, setOrder] = useState(true);
   const [searchParams, setSearchParams] = useSearchParams();
-  const filterByTopic = searchParams.get('topic');
+  const filterByTopic = searchParams.get("topic");
+  const sortBy = searchParams.get("sort_by");
+  const orderBy = searchParams.get("order");
 
+  const setOrderParams = (order) => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("order", order);
+    setSearchParams(newParams);
+  };
+
+  const setSortByParams = (sortBy) => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("sort_by", sortBy);
+    setSearchParams(newParams);
+  };
+  $(document).ready(function() {
+    $('.ui.dropdown').dropdown();
+  });
   useEffect(() => {
-      fetchArticles(filterByTopic).then((data) => {
+    
+    fetchArticles(filterByTopic, sortBy, orderBy).then((data) => {
       setArticles(data.articles);
+      setSortAttributes(["created_at", "votes", "comment_count"]);
       setIsLoading(false);
     });
-  }
-  , [filterByTopic]);
+  }, [filterByTopic, sortBy, orderBy]);
 
   if (isLoading) return <Loading />;
 
   return (
-    <div className="ui three stackable link cards">
-      {articles.map((article) => {
-        return <Articles key={article.article_id} article={article} />;
-      })}
-    </div>
+    <>
+      <div className="ui top attached menu">
+        <div className="right menu">
+          <div className="ui right dropdown item">
+            <div className="text">Sort By</div>
```

Comment:
definitely would consider throwing in some semantic html tags wherever possible to avoid a sea of divs

Reply:
I'm planning to do this more towards the end because I am working with a library that I am not totally familiar and also my CSS skills are not strong right now.